pallets/streaming/src/lib.rs

```diff
@@ -381,65 +396,46 @@ impl<T: Config> Pallet<T> {
         T::PalletId::get().into_account()
     }
 
-    pub fn delta_of(stream: &Stream<T>) -> Result<u64, DispatchError> {
-        let now = T::UnixTime::now().as_secs();
-        if now <= stream.start_time {
-            Ok(Zero::zero())
-        } else if now < stream.stop_time {
-            now.checked_sub(stream.start_time)
-                .ok_or(DispatchError::Arithmetic(ArithmeticError::Underflow))
-        } else {
-            stream
-                .stop_time
-                .checked_sub(stream.start_time)
-                .ok_or(DispatchError::Arithmetic(ArithmeticError::Underflow))
-        }
+    pub fn update_finished_stream_library(account: &AccountOf<T>) -> DispatchResult {
+        let checked_pop =
+            |regisry: &mut Option<BoundedVec<StreamId, T::MaxStreamsCount>>| -> DispatchResult {
+                let mut r = regisry.take().unwrap_or_default();
+                r.as_mut().sort_unstable_by(|a, b| b.cmp(a));
+
+                let len = r.len() as u32;
+                match len {
+                    _x if len >= T::MaxFinisehdStreamsCount::get() => {
+                        if let Some(index) = r.pop() {
```

Comment:
Remove the sort_unstable_by and use r.first() ?

Reply:
<img width="329" alt="image" src="https://user-images.githubusercontent.com/44558375/167260186-f1cf9a06-a505-42f6-b2af-73c2e039d501.png">

Reply:
🤔